src/core/sagas/recommendations.ts

```diff
@@ -49,56 +59,110 @@ export namespace Tasks {
 
         yield effects.put(flux.actions.receiveRecommendationsProducts(SearchAdapter.augmentProducts(results)));
       }
-      return [];
     } catch (e) {
       yield effects.put(flux.actions.receiveRecommendationsProducts(e));
     }
   }
 
-  export function* fetchPastPurchases(flux: FluxCapacitor, { payload }: Actions.FetchPastPurchases) {
+  export function* fetchSkus(config: Configuration, endpoint: string, query?: string) {
+    const securedPayload = ConfigAdapter.extractSecuredPayload(config);
+    if (securedPayload) {
+      const url = `https://${config.customerId}.groupbycloud.com/orders/v1/public/skus/${endpoint}`;
+      const response = yield effects.call(utils.fetch, url, Adapter.buildBody({
+        securedPayload,
+        query
+      }));
+      return yield response.json();
+    }
+    throw new MissingPayload();
+  }
```

Comment:
why is creating new error message so complicated? Can't we do just throw new Error('missing payload')?